codex/sales/states/provingsimulated.nim

```diff
@@ -32,8 +33,8 @@ when codex_enable_proof_failures:
       try:
         warn "Submitting INVALID proof", period = currentPeriod, slotId = slot.id
         await market.submitProof(slot.id, newSeq[byte](0))
-      except ProviderError as e:
-        if not e.revertReason.contains("Invalid proof"):
+      except SignerError as e:
```

Comment:
Looking at this again, it could be a `ProviderError` or `SignerError` (`EstimateGasError` is derived from `SignerError`). But, you're correct, this should probably be a `MarketError`.

For now, I've backed out this change as I believe it was from quite some time ago trying to filter error messages from `EstimateGasError`. After your comment prompted me to re-look at the exception handling, I feel there are further changes needed, for consistency, and they should be scoped to a new PR.

Reply:
Above changes in https://github.com/codex-storage/nim-codex/pull/670